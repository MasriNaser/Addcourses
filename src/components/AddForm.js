import React from 'react';
import { Button, Form } from 'semantic-ui-react';
import PropTypes from 'prop-types';
const AddForm = props => {
  return (
    <Form onSubmit={props.ookNaarForm}>
      <input value={props.currentEmpty} type='text' onChange={props.naarForm} 
      placeholder='Add a course here...'
      autoFocus
      />
      <Button style={buttonStyle} type='submit'>
        Please add a course
      </Button>
    </Form>
  );
};
//propTypes
AddForm.propTypes = {
  naarForm: PropTypes.func.isRequired,
  currentEmpty: PropTypes.string.isRequired
};
//styling

const buttonStyle = {
  margin: '10px 44%'
};

export default AddForm;
