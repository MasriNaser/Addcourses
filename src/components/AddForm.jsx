import React from 'react';
import { Button, Form } from 'semantic-ui-react';
import PropTypes from 'prop-types';
const AddForm = props => {
  const {naarForm,ookNaarForm,currentEmpty} = props
  return (
    <Form onSubmit={ookNaarForm}>
      <input value={currentEmpty} type='text' onChange={naarForm} 
      placeholder='Add a course here...'
      autoFocus
      />
      <Button style={buttonStyle} type='submit'>
        Please add a course
      </Button>
    </Form>
  );
};
//propTypes
AddForm.propTypes = {
  naarForm: PropTypes.func.isRequired,
  currentEmpty: PropTypes.string.isRequired
};
//styling

const buttonStyle = {
  margin: '10px 44%'
};

export default AddForm;
