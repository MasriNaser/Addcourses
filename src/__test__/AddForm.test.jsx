import React from 'react';
import { shallow } from 'enzyme';
import AddForm from '../components/AddForm';
import toJson from 'enzyme-to-json';

const onClick = jest.fn().mockName('onClick');
const wrapper = shallow(
  <AddForm updateForm={onClick} currentEmpty='' buttonName='' />
);
describe('AddForm component', () => {
  it('should take snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should has Form', () => {
    expect(wrapper.find('Form').exists()).toBe(true);
  });
  it('should has input', () => {
    expect(wrapper.find('input').exists()).toBe(true);
  });
  it('should has button', () => {
    expect(wrapper.find('Button').exists()).toBe(true);
  });
});
//test if the components is passed to app.js
