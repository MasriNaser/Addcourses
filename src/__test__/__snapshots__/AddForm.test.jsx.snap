// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm component should take snapshot 1`] = `
<Form
  as="form"
  date-test="dataForm"
>
  <input
    autoFocus={true}
    date-test="inputData"
    onChange={[MockFunction onClick]}
    placeholder="Add a course here..."
    type="text"
    value=""
  />
  <Button
    as="button"
    data-test="buttonData"
    style={
      Object {
        "margin": "10px 44%",
      }
    }
    type="submit"
  />
</Form>
`;
